src/material-experimental/mdc-input/harness/input-harness.spec.ts

```diff
@@ -0,0 +1,207 @@
+      loader = TestbedHarnessEnvironment.loader(fixture);
+      inputHarness = MatInputHarness;
+    });
+
+    runTests();
+  });
+
+  describe(
+      'MDC-based',
+      () => {
+          // TODO: run tests for MDC based input once implemented.
+      });
+});
+
+/** Shared tests to run on both the original and MDC-based input's. */
+function runTests() {
+  it('should load all input harnesses', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+  });
+
+  it('should load input with specific id', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness.with({id: 'myTextarea'}));
+    expect(inputs.length).toBe(1);
+  });
+
+  it('should load input with specific name', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness.with({name: 'favorite-food'}));
+    expect(inputs.length).toBe(1);
+  });
+
+  it('should load input with specific value', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness.with({value: 'Sushi'}));
+    expect(inputs.length).toBe(1);
+  });
+
+  it('should be able to get id of input', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+    expect(await inputs[0].getId()).toMatch(/mat-input-\d+/);
+    expect(await inputs[1].getId()).toMatch(/mat-input-\d+/);
+    expect(await inputs[2].getId()).toBe('myTextarea');
+  });
+
+  it('should be able to get name of input', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+    expect(await inputs[0].getName()).toBe('favorite-food');
+    expect(await inputs[1].getName()).toBe('');
+    expect(await inputs[2].getName()).toBe('');
+  });
+
+  it('should be able to get value of input', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+    expect(await inputs[0].getValue()).toBe('Sushi');
+    expect(await inputs[1].getValue()).toBe('');
+    expect(await inputs[2].getValue()).toBe('');
+  });
+
+  it('should be able to set value of input', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+    expect(await inputs[0].getValue()).toBe('Sushi');
+    expect(await inputs[1].getValue()).toBe('');
+
+    await inputs[0].setValue('');
+    await inputs[2].setValue('new-value');
+
+    expect(await inputs[0].getValue()).toBe('');
+    expect(await inputs[2].getValue()).toBe('new-value');
+  });
+
+  it('should be able to get disabled state', async () => {
+    const inputs = await loader.getAllHarnesses(inputHarness);
+    expect(inputs.length).toBe(3);
+
+    expect(await inputs[0].isDisabled()).toBe(false);
+    expect(await inputs[1].isDisabled()).toBe(false);
+    expect(await inputs[2].isDisabled()).toBe(false);
+
+    fixture.componentInstance.disabled = true;
+    fixture.detectChanges();
```

Comment:
I'm trying to think if it should be best practice to leave it up to the harness though. Regardless of the harness in my opinion the `detectChanges` should be called to make the property change in the component instance effective.

Reply:
I'm not sure what we should do in case of the harness tests though. It somehow feels weird to me that we remove the `detectChanges`. It would make sense to me to remove it, if the tests are also generic and run against _both_  protractor and testbed.

Reply:
It's certainly not going to _hurt_ to have it here. But the harness automatically calls `fixture.detectChanges` and `await fixture.whenStable()` before reading any attribute or property. So I think as a stylistic thing for tests that rely on harnesses, we should probably not add this unless its needed for some reason (and in that case we would probably want to document why its needed)

Reply:
I could see the point of removing it in favor of ensuring that it technically works that the harness automatically detects changes/stabilizes. We should try to document the reasons/style guidelines for writing harness tests though IMO.

Updated! :smile: